src/sentry/incidents/endpoints/serializers/alert_rule.py

```diff
@@ -455,7 +467,7 @@ def serialize(
             updated_attrs["type"] = "alert_rule"
         elif isinstance(obj, Rule):
             updated_attrs["type"] = "rule"
-        elif isinstance(obj, ProjectUptimeSubscription):
+        elif isinstance(obj, Detector) and obj.type == GROUP_TYPE_UPTIME_DOMAIN_CHECK_FAILURE:
```

Comment:
This is also not a great pattern... do you know if this is a long term pattern or just transitional? People shouldn't need to add things like this in a bunch of places, it should just be on the grouptype

Reply:
I think this is _just_ because of what we're doing here in this combined alert serializer, which would go away in the future right?